src/libs/Navigation/linkTo.js

```diff
@@ -66,7 +67,10 @@ export default function linkTo(navigation, path, type) {
             action.type = CONST.NAVIGATION.ACTION_TYPE.REPLACE;
 
             // If this action is navigating to the report screen and the top most navigator is different from the one we want to navigate - PUSH the new screen to the top of the stack
-        } else if (action.payload.name === NAVIGATORS.CENTRAL_PANE_NAVIGATOR && getTopmostReportId(root.getState()) !== getTopmostReportId(state)) {
+        } else if (
+            action.payload.name === NAVIGATORS.CENTRAL_PANE_NAVIGATOR &&
+            (getTopmostReportId(root.getState()) !== getTopmostReportId(state) || getTopMostCentralPaneRouteName(root.getState()) !== getTopMostCentralPaneRouteName(state))
```

Comment:
@adamgrzybowski could you extract this conditional statement into a variable and add a comment explaining what you're doing on this inline?